### DISCUSSION TITLE
Sonoff Mini con Tasmota in Automation

### DISCUSSION THREAD

**Discussion by polinter** [indomus/forum#295] [HUB: Home Assistant]
Buongiorno a tutti. Ho installato con successo un Sonoff mini utilizzando il firmware tasmota.

L'ho integrato con successo in Home Assistant con il seguente codice: 

`switch:
  platform: command_line
  switches:
    my_sonoff_switch:
      command_on:      "/usr/bin/curl -X POST http://192.168.1.4/cm?cmnd=Power%20On"
      command_off:     "/usr/bin/curl -X POST http://192.168.1.4/cm?cmnd=Power%20Off"
      command_state:   "/usr/bin/curl -X GET http://192.168.1.4/cm?cmnd=Power"
      value_template:  '{{value_json.POWER == "ON" }}'
      friendly_name:   "Luce Bagno"  `

Adesso avrei bisogno di fare in modo di richiamarlo all'interno di alcune automazioni. Ad esempio vorrei fare in modo che la luce si accendesse quando si apre una porta su cui ho posizionato un sensore di contatto.

Qual'è il comando da utilizzare per dare il comando di accensione/spegnimento di uno switch così creato?

Grazie in anticipo a tutti coloro che mi daranno una dritta.

**Comment by KC-inDomus** [comment]
scusa,
domanda,
perché non usi MQTT?

**Reply by polinter** [reply]
Ho provato in tutti i modi. Con mosquitto mqtt e con Tasmoadmin non rileva il sonoff.

Dall'interfaccia del Tasmota risulta regolarmente connesso al mqtt ma poi in home assistant non c'è verso di farlo visualizzare.

**Reply by KC-inDomus** [reply]
dunque c'è un problema lato HA, o in termini di integrazione MQTT o in termini di configurazione del sonoff via MQTT

**Reply by polinter** [reply]
Per certo il sonoff è configurato correttamente. Dalla console Tasmota ricevo tutti i messaggi di connessione avvenuta con successo con l'host che ho impostato (che poi è l'ip di HA). In HA però poi non riesco a visualizzare il dispositivo.

Non viene semplicemente rilevato...

**Reply by KC-inDomus** [reply]
ripeto: quindi il problema è lato HA

**Comment by snakuzzo** [comment]
Se parli del comando da utilizzare su HA, avendo creato un'entità del dominio `switch`, devi utilizzare il servizio `switch.turn_on`.
Questo un esempio di automazione per accendere la luce al passaggio ad `'on'` di un sensore binario:


```yaml
- id: mia_automazione
  alias: "Mia Automazione"
  trigger:
    - platform: state
      entity_id: binary_sensor.mio_sensore
      to: 'on'
  action:
    - service: switch.turn_on
      data:
        entity_id: switch.my_sonoff_switch
```

**Reply by polinter** [reply]
Grazie mille era proprio quello che mi serviva.
